boardgame-ui/src/components/event/EventAddEditGame.vue

```diff
@@ -0,0 +1,314 @@
+      for(let link of this.boardGamesLinks) {
+        let id = link.id_board_game;
+        if(!processedIds.includes(id)) {
+          boardGames.push(link.provided_board_game);
+          processedIds.push(id);
+        }
+      }
+      return boardGames;
+    },
+    filteredBoardGames() {
+      if(!this.searchString) {
+        return this.boardGames;
+      }
+
+      let str = this.searchString.toLowerCase();
+      return this.boardGames.filter(bg => bg.name.toLowerCase().indexOf(str) >= 0);
+    },
+    allowedRankingMethods() {
+      return [GameRankingMethods.WIN_LOSE, GameRankingMethods.POINTS_HIGHER_BETTER];
+    },
+    ranked() {
+      return this.game.ranking_method === GameRankingMethods.POINTS_HIGHER_BETTER;
+    },
+    users() {
+      return this.event.attendees.map(attendee => attendee.user);
+    },
+    selectedUsersIds() {
+      return this.players.map(({user}) => user ? user.id : 0);
+    }
+  },
+  watch: {
+    'game.ranking_method'(_, old) {
+      if(!old) {
+        return;
+      }
+      this.players.forEach(player => player.score = null); // reinitialize scores
+    },
+    time() {
+      if(this.time < this.minTime) {
+        this.time = this.minTime;
+      }
+    }
+  },
+  methods: {
+    selectBoardGame(option) {
+      this.game.id_board_game = option ? option.id : null;
+    },
+    addPlayer() {
+      this.players.push({user: null, score: null});
+    },
+    removePlayer(idx) {
+      if(this.players.length === 1) {
+        this.$toast.open({
+          message: this.$t('add-edit-game.must-have-at-least-one-player'),
+          type: 'is-danger',
+          position: 'is-bottom'
+        });
+        return;
+      }
+      this.players.splice(idx, 1);
+    },
+    async save() {
+      let result = await this.$validator.validateAll();
+
+      if(!this.game.id_board_game) {
+        this.$validator.errors.add({field: 'boardGame', msg: this.$t('add-edit-game.validation-error.board-game')});
+        result = false;
+      }
+
+      if(!result) {
+        this.$toast.open({
+          message: this.$t('global.invalid-form'),
+          type: 'is-danger',
+          position: 'is-bottom'
+        });
+        return;
+      }
+
+      this.game.duration = this.time.getHours()*60 + this.time.getMinutes();
+      this.game.players = this.players.map(({user, score}) => ({user: user.id, score: Number(score)}));
+
+      try {
+        await this.game.save();
+        this.$toast.open({
+          message: this.$t('add-edit-game.save-success'),
+          type: 'is-success',
+          position: 'is-bottom'
+        });
+        this.$emit('addGame', this.game);
+      }
+      catch(error) {
+        console.log(error);
+        this.$toast.open({
+          message: this.$t('add-edit-game.save-error'),
+          type: 'is-danger',
+          position: 'is-bottom'
+        });
+      }
+    }
+  },
+  async created() {
+    this.boardGamesLinks = await this.event.fetchBoardGames();
```

Comment:
why "boardGamesLinks" as variable name and not simply "boardGames" ?

Reply:
Because if several people bring the same board game to the event, we will have several board games links. See board games computed property. Or maybe there is another endpoint in backend allowing to retrieve directly the board games of the event, rather than the (board game-user) couples ?

Reply:
Ok, I understand now. I am calling such object a `ProvidedBoardGame` in the back-end but I am not totally satisfied with this naming. `boardGamesLink` is fine then !